docs: fix hyperlinks and other minor issues

qmonnet opened the pull request:
This is a fix for hyperlinks under `Documentation/`, and for other (mostly) related issue. I ran [a tool](https://github.com/dkhamsing/awesome_bot) to check links on documentation, and found a number of dead link. This tool also reported links pointing to redirections, or duplicate links in source files. Let's try to fix all of this:

- Fix broken links (404)
- Fix links pointing to wrong targets
- Use final destination for (most) links pointing to redirections (two commits, one is just for `docs.cilium.io` in file headers)
- Avoid duplicated links as much as possible, use names hyperlinks instead
- While at it, replace references to “covalent” in links, let's avoid any confusion for the readers

The STAR WARS API used in the documentation, tests and examples for TLS visibility has been taken down. Let's replace it with something else (https://artii.herokuapp.com), cc  @danwent.

Additional miscellaneous fixes:

- Fix a typo in CI doc (`ct/next-next`, sounds funny but is unused)
- Fix retrieval of current release number in Python script, due to break some day otherwise
- Add `test-me-please` to the list of keywords triggering GKE tests in CI
- Move URL and update name for `Cilium-Nightly-Tests Job`, fix link for Flannel job, in CI doc. **Please double check this commit, I am not 100% sure the change is correct.**

After this PR, I still observe **broken links for bisects on Jenkins** (CI doc). I am unsure if the links are broken or if I simply lack permission to access them, can someone please check and let me know?

pchaigno commented on test/k8sT/Policies.go:
```diff
@@ -325,14 +325,14 @@ var _ = Describe("K8sPolicyTest", func() {
 				namespaceForTest, appPods[helpers.App2],
-				helpers.CurlFail("--retry 5 -4 %s https://swapi.co:443/api/planets/2/", "-v --cacert /cacert.pem"))
-			res.ExpectFailWithError("403 Forbidden", "Unexpected connection from %q to 'https://swapi.co:443/api/planets/2/'",
+				helpers.CurlFail("--retry 5 -4 --max-time 15 %s 'https://artii.herokuapp.com:443/fonts_list'", "-v --cacert /cacert.pem"))
```
Why did you remove the `--max-time` parameter above and add it here?

qmonnet replied:
Uh, that was not intentional. Thanks (So I replaced the two URLs, realised that I had swapped the one supposed to work with the one supposed to fail, and swapped the lines without noticing the `--max-time` difference). I'll fix.

pchaigno commented on Documentation/conf.py:
```diff
@@ -91,18 +91,19 @@
 archive_name = 'cilium-' + archive_name.strip('v')
 project_link = 'https://github.com/cilium/cilium/projects?query=is:open+' + next_release
-backport_format = 'https://github.com/cilium/cilium/pulls?q=is:open+is:pr+label:%s/' + current_release
+backport_format = 'https://github.com/cilium/cilium/pulls?q=is:open+is:pr+label:backport/%s' + current_release
```
I _think_ this is incorrect. [Those links](https://docs.cilium.io/en/latest/contributing/release/backports/#preparation) already work. And according to [the Sphinx documentation](https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html), the `%s` here is to be replaced in the reStructuredText document, not by `current_release`.

qmonnet replied:
Oh I missed that, thanks for checking! I'll fix.

pchaigno commented on test/k8sT/manifests/l7-policy-TLS.yaml:
```diff
@@ -16,23 +16,23 @@ spec:
         http:
         - method: "GET"
-          path: "/api/planets/1/"
+          path: "/make??.*"
```
As discussed offline, the `??` here seems incorrect.

pchaigno reviewed (changes requested):
Couple nits below, but looks good overall! Thanks for taking care of all those links :link: !

qmonnet committed:
docs: move URL for "Cilium-Nightly-Tests Job", fix links for CI doc

Big source files for documentation occasionally get split out into
several pieces. At one occasion, the address for hypertext links on
"Cilium-Nightly-Tests Job" was not moved to the same file as the
reference itself, thus breaking the links. Sphinx apparently complained
and the references were later turned into simple preformatted chunks.

Move the URL to the relevant file (it is unused in its current file
anyway) and restore the links.

Also update the name and target for this link, and the target for the
Flannel job.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: fix links pointing to redirections

As an attempt to clean up documentation, fix most of the links for which
content has been moved (301 or 302 redirections). No need to have
readers make more HTTP(S) requests than necessary.

Redirections are preserved when redirection target is too ugly and not
intended to be called directly, or when we use a redirection on purpose
(e.g. "/latest/").

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: fix current_release number retrieval

Getting the current release number by extracting the first four
characters from the string read from file VERSION is gonna bite sooner
or later. Let's get the proper major and minor fields instead,
especially because we already parsed them a few lines above.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs, test: replace swapi by artii.herokuapp.com for TLS visibility

The STAR WARS API used as an example for TLS visibility documentation
and tests is no longer maintained [0], and the website is down.

Let's switch to something else instead: https://artii.herokuapp.com/,
which turns arguments into ASCII art snippets.

[0] https://github.com/phalt/swapi

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: replace references to covalent in links

Covalent has been renamed, let's remove its occurrences from links in
documentation to avoid confuse the readers.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: try to avoid duplicate links

Try to remove duplicate occurrences of link URLs, and use named
hyperlinks wherever possible instead, in order to ease readability of
RST files and maintenance.

Some duplicates are preserved, for example when they are in examples and
code snippets and it would not make sense to replace them.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: fix typo on flag for running on net-next

Typo in documentation: "ci/next-next". Fix to "ci/net-next".

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: fix bad links (wrong target)

Fix two links that do point to valid pages, but not the ones we expect.

Jesper's presentation does not have the correct link, it looks like a
copy-paste error.

The documentation for Minikube has moved. There is a redirection in
place, but we land on a higher level “getting started” page. Let's go
directly to the Minikube page instead.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: fix broken links

Fix several broken links in documentation.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: add test-me-please as keyword to trigger GKE tests

GKE tests can be triggered with test-gke, but also now with
test-me-please. Update CI documentation accordingly.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
docs: use HTTPS for links to docs.cilium.io

Use HTTPS instead of HTTP for links to docs.cilium.io in headers for
documentation source files.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>
